cmd/controller-manager/app/controller-manager.go

```diff
@@ -107,6 +109,22 @@ func Run(opts *options.Options, stopChan <-chan struct{}) error {
 		panic(err)
 	}
 
+	runningInCluster := len(masterURL) == 0 && len(kubeconfig) == 0
+	if runningInCluster && len(opts.Config.KubeFedNamespace) == 0 {
+		// For in-cluster deployment set the namespace associated
+		// with the service account token
+		data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+		if err != nil {
+			klog.Fatalf("An error occurred while attempting to discover the kubefed namespace from the service account: %v", err)
+		}
+		opts.Config.KubeFedNamespace = strings.TrimSpace(string(data))
+	}
+
+	// Validate if a kubefed-namespace is configured
+	if len(opts.Config.KubeFedNamespace) == 0 {
+		klog.Fatalf("The kubefed namespace must be specified via --kubefed-namespace")
```

Comment:
nit: Please prefer KubeFed in output seen by the user.

Reply:
Done.